### PR TITLE
ODPR-1099: Update Frontend Support For Registration With ID.

### PR DESCRIPTION
**Pull request by paddynski-opencast**

### CODE REVIEW

File: it/test/services/registration/withId/RegistrationWithIdForIndividualServiceSpec.scala [hunk 1]
```diff
@@ -384,18 +478,183 @@ class RegistrationWithIdForIndividualServiceSpec extends BaseBackendConnectorSpe
...
+                  CommonResponses.Error("eis-returned-unauthorised")
+                )
+              )
+            )
             verifyThatDownstreamApiWasCalled()
           }
         }
       }
+      "invalid, with a status code of" - {
+        "service unavailable" in {
```
Comment (erasmos-hmrc-new) [inline it/test/services/registration/withId/RegistrationWithIdForIndividualServiceSpec.scala:551]: Given that the backend will no longer reply with this error code, could you change it to the 'internal error' case?

File: it/test/services/registration/withId/RegistrationWithIdForOrganisationServiceSpec.scala [hunk 2]
```diff
@@ -257,10 +354,153 @@ class RegistrationWithIdForOrganisationServiceSpec extends BaseBackendConnectorS
...
+              _type = Requests.Type.CorporateBody
+            )
+
+            val response = await(service.call(request))
+
+            response shouldBe Left(
+              CommonResponses.Errors(
+                UNAUTHORIZED,
+                Seq(
+                  CommonResponses.Error("eis-returned-unauthorised")
```
Comment (erasmos-hmrc-new) [inline it/test/services/registration/withId/RegistrationWithIdForOrganisationServiceSpec.scala:503]: Given that the backend will no longer reply with this error code, could you change "invalid, with a status code of / service unavailable" to the 'internal error' case?

**Review by erasmos-hmrc-new — CHANGES_REQUESTED** [review]

File: it/test/services/registration/withId/RegistrationWithIdForIndividualServiceSpec.scala (hunk 1 above)
Comment (paddynski-opencast) [reply it/test/services/registration/withId/RegistrationWithIdForIndividualServiceSpec.scala]: Changed.

File: it/test/services/registration/withId/RegistrationWithIdForOrganisationServiceSpec.scala (hunk 2 above)
Comment (paddynski-opencast) [reply it/test/services/registration/withId/RegistrationWithIdForOrganisationServiceSpec.scala:503]: Changed.

File: it/test/services/registration/withId/RegistrationWithIdForIndividualServiceSpec.scala (hunk 1 above)
Comment (erasmos-hmrc-new) [reply it/test/services/registration/withId/RegistrationWithIdForIndividualServiceSpec.scala]: Confirmed.

File: it/test/services/registration/withId/RegistrationWithIdForOrganisationServiceSpec.scala (hunk 2 above)
Comment (erasmos-hmrc-new) [reply it/test/services/registration/withId/RegistrationWithIdForOrganisationServiceSpec.scala:503]: Confirmed.

**Review by erasmos-hmrc-new — APPROVED** [review]

**Pull request merged**